pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,379 @@
+  
+  - in: one of the following values: `query`, `header`, `path`, `formData`, `body`
+  - type: one of the following values:  `string`, `number`, `boolean`, `integer`
+  - name: a `string`, name of the parameter
+
+  So an example would be `@param.query.number('offset')`. 
+  You can find the specific usage in [Writing Controller methods](controller.htm#writing-controller-methods)
+  
+## Dependency Injection
+
+`@inject` is a decorator to annotate method arguments for automatic injection by 
+LoopBack IoC container. `@inject` can only be used on properties or method parameters.
+
+The `@inject` decorator allows you inject dependencies bound to the application context, 
+to the per-request contextor, to any implementation of the [Context](#context) object. 
+You can bind values, class definitions and provider functions to those contexts and 
+then resolve values (or the results of functions that return those values!) in other 
+areas of your code.
+
+```ts
+// application.ts
+import {Application} from '@loopback/core';
+import 'fs-extra';
+class MyApp extends Application {
+  constructor() {
+    super();
+    const app = this;
+    const widgetConf = JSON.parse(fs.readSync('./widget-config.json'));
+    app.bind('config.widget').to(widgetConf);
+  }
+}
+```
+
+Now that we've bound the 'config.widget' key to our configuration object, we can
+inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  constructor(
+    @inject('config.widget') protected widget: JSON
+    // This will be resolved at runtime!
+  ) {}
+  // etc...
+}
+```
+
+For more information, see the [Dependency Injection](Dependency-Injection.htm) section under [LoopBack Core Concepts](Concepts.htm)
+
+## Authentication Decorator
+
+  Syntax: `@authenticate(strategyName: string, options?: Object)`
+
+  Marks a controller method as needing an authenticated user. 
+  This decorator requires a strategy name as a parameter.
+
+  Here's an example using 'BasicStrategy': to authenticate user in function `whoAmI`:
+
+  ```ts
+  // my-controller.ts
+  import { authenticate } from '@loopback/authentication';
+  import { inject } from '@loopback/context';
+
+  class MyController {
+    constructor(
+      @inject(BindingKeys.Authentication.CURRENT_USER)
+      private user: UserProfile,
+    ) {}
+
+    @authenticate('BasicStrategy')
+    async whoAmI() : Promise<string> {
+      return this.user.id;
+    }
+  }
+  ```
+
+## Repository Decorators
+
+  As a Domain-driven design concept, 
+  repository means a mediate between your domain object and data mapping layers
+  using a collection-like interface for accessing domain objects. 
+  In framework LoopBack, a domain object is usually a TypeScript/JavaScript Class instance, 
+  and a typical example of data mappting layer module could be a database's node.js driver.
```

Comment:
`...a typical example of data mappting...` -> `...a typical example of a data mapping...`